package/kedro_viz/api/responses.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2021 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""`kedro_viz.api.responses` defines API response types"""
+# pylint: disable=missing-class-docstring,too-few-public-methods
+import abc
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel
+
+from kedro_viz.data_access import data_access_manager
+
+
+class APIErrorMessage(BaseModel):
+    message: str
+
+
+class BaseAPIResponse(BaseModel, abc.ABC):
+    class Config:
+        orm_mode = True
+
+
+class BaseGraphNodeAPIResponse(BaseAPIResponse):
+    id: str
+    name: str
+    full_name: str
+    tags: List[str]
+    pipelines: List[str]
+    modular_pipelines: List[str]
+    type: str
+
+
+class TaskNodeAPIResponse(BaseGraphNodeAPIResponse):
+    parameters: Dict
+
+    class Config:
+        schema_extra = {

```

Comment:
What's this used for? Is it just for the API swagger/redoc docs or is it some sort of schema check as well?

Reply:
Just to show up in the swagger doc as exmaple API response :D